fsspec/utils.py

```diff
@@ -307,7 +307,11 @@ def stringify_path(filepath):
         return filepath.__fspath__()
     elif isinstance(filepath, pathlib.Path):
         return str(filepath)
-    return filepath
+    else:
+        try:
+            return str(filepath)
+        except Exception:
+            return filepath
```

Comment:
I'm unsure about what situations returning `filepath` unchanged is desired. I'd like someone with a bit more understanding of that to weigh in on that.